api/routes/analysis.py

```diff
@@ -106,36 +119,52 @@ async def get_incomplete_analyses(
     return results
 
 
+@router.get(
+    '/{project}/{analysis_type}/latest-complete',
+    operation_id='getLatestCompleteAnalysisForType',
+)
+async def get_latest_complete_analysis_for_type(
+    analysis_type: AnalysisType,
+    connection: Connection = get_project_readonly_connection,
+):
+    """Get latest complete analysis for some analysis type"""
+    alayer = AnalysisLayer(connection)
+    analysis = await alayer.get_latest_complete_analysis_for_type(
+        project=connection.project, analysis_type=analysis_type
+    )
+    if analysis:
+        analysis.sample_ids = sample_id_format(analysis.sample_ids)
+
+    return analysis
+
+
 @router.post(
-    '/latest_complete/{analysis_type}/for-samples',
-    operation_id='getLatestAnalysisForSamplesAndType',
+    '/{project}/{analysis_type}/latest-complete',
+    operation_id='getLatestCompleteAnalysisForTypePost',
 )
-async def get_latest_analysis_for_samples_and_type(
-    sample_ids: List[str],
+async def get_latest_complete_analysis_for_type_post(
     analysis_type: AnalysisType,
-    allow_missing: bool = True,
+    meta: Dict[str, Any] = Body(..., embed=True),
     connection: Connection = get_project_readonly_connection,
 ):
-    """Get latest complete gvcfs for sample"""
-    atable = AnalysisLayer(connection)
-    results = await atable.get_latest_complete_analysis_for_samples_and_type(
+    """Get latest complete analysis for some analysis type"""
+    alayer = AnalysisLayer(connection)
+    analysis = await alayer.get_latest_complete_analysis_for_type(
+        project=connection.project,
         analysis_type=analysis_type,
-        sample_ids=sample_id_transform_to_raw(sample_ids),
-        allow_missing=allow_missing,
+        meta=meta,
     )
+    if analysis:
+        analysis.sample_ids = sample_id_format(analysis.sample_ids)
 
-    if results:
-        for result in results:
-            result.sample_ids = sample_id_format(result.sample_ids)
-
-    return results
+    return analysis
 
 
 @router.get(
     '/{project}/{analysis_type}/latest-complete',
     operation_id='getLatestCompleteAnalysisForType',
 )
-async def get_latest_complete_analysis_for_type(
+async def get_latest_complete_analysis_for_type_meta(
```

Comment:
What's the difference between `get_latest_complete_analysis_for_type_meta` and `get_latest_complete_analysis_for_type`

Reply:
Nothing, this was incorrectly refactored (and I've renamed back).